kratos/utilities/tessellation_utilities/tessellation.h

```diff
@@ -0,0 +1,70 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+
+#if !defined(KRATOS_TESSELLATION_H_INCLUDED )
+#define  KRATOS_TESSELLATION_H_INCLUDED
+
+namespace Kratos {
+
+template <class TContainerPointType>
+class Tessellation
+{
+public:
+    typedef Geometry<typename TContainerPointType::value_type> GeometryType;
+    typedef typename GeometryType::CoordinatesArrayType CoordinatesArrayType;
+
+    ///@name Constructors
+    ///@{
+
+    /// Conctructor for tessellation
+    Tessellation()
+    {
+    }
+
+    /// Copy Constructor
+    Tessellation(Tessellation<TContainerPointType> const& rOther)
+    {
+    }
+
+    /// Assignment Operator
+    Tessellation<TContainerPointType>& operator=(const Tessellation<TContainerPointType>& rOther)
+    {
+        return *this;
+    }
+
+    ///@}
+    ///@name Constructors
+    ///@{
+
+    virtual void Tessellate(
+        const GeometryType& rGeometry,
+        const double Tolerance,
+        const int NumberOfGuessesPerInterval = 1)
+    {
+        KRATOS_ERROR << "Calling Tessellate from base Tessellation class."
```

Comment:
Are you going to consider the Delaunator utilities  (https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/delaunator_utilities.cpp) in a derived class?

Reply:
Yes, I actually want to take this into account for 2D.

Reply:
:+1: